### PR TITLE
add refute_requested

### PR DESCRIPTION
**Pull request by grosser**
@bblimke @irwaters

### CODE REVIEW

File: test/shared_test.rb [hunk 1]
```diff
@@ -22,7 +22,6 @@ def test_assert_not_requested_with_stub_and_block_raises_error
   end
 
   def test_error_on_non_stubbed_request
-    default_ruby_headers = (RUBY_VERSION >= "1.9.1") ? "{'Accept'=>'*/*', 'User-Agent'=>'Ruby'}" : "{'Accept'=>'*/*'}"
```
Comment (grosser) [inline test/shared_test.rb]: was not used

File: test/shared_test.rb [hunk 2]
```diff
@@ -48,7 +47,7 @@ def test_verification_that_expected_request_didnt_occur
     end
   end
 
-  def test_verification_that_expected_request_didnt_occur
+  def test_verification_that_expected_stub_didnt_occur
```
Comment (grosser) [inline test/shared_test.rb]: had the same name as the test above ... so never got executed

**Comment by bblimke** [comment]
@grosser thanks. is `refute` a popular term when it comes to asserting? Do any test frameworks use it?  Perhaps it is. I'm asking because I've never seen `refute` in any test DSL.

**Comment by grosser** [comment]
Minitest ships with it and imo it's the nicer alternative to assert_not which reads weird and does not align

https://github.com/seattlerb/minitest/search?utf8=%E2%9C%93&q=refute

and using it a bunch here:
https://github.com/zendesk/samson/search?utf8=%E2%9C%93&q=refute

**Comment by bblimke** [comment]
@grosser excellent. Thanks!

**Pull request merged**